alf/environments/wrappers.py

```diff
@@ -12,187 +12,166 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import collections
-from collections import OrderedDict, deque
-from copy import deepcopy
-
-import gin
+import copy
+from collections import deque
 import gym
 import numpy as np
 import cv2
-
-from absl import logging
-
+import gin
 from tf_agents.environments import wrappers
 from tf_agents.trajectories.time_step import StepType
+from alf.utils import common
+
+
+def traverse_transform_space(space, levels, field, func):
+    """Transform space
```

Comment:
Need more explanation on what this function does: given the argument `space`, `levels`, `field` and `func` what kind of output it generates?

Reply:
A more concrete comment: Transform the child space in `observation_space` indicated by `field` using `func`

Reply:
updated